frontend/src/components/common/PrintCSVButton.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import {
+  Icon,
+  IconButton,
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  FormControl,
+  Button,
+  FormHelperText,
+  Flex,
+  ScaleFade,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import { AiFillPrinter } from "react-icons/ai";
+import { RangeDatepicker } from "chakra-dayzed-datepicker";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import CSVConverter from "../../helper/CSVConverter";
+
+const PrintCSVButton = (): React.ReactElement => {
+  const [selectedDates, setSelectedDates] = useState<Date[]>([]);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handleClear = () => {
+    setSelectedDates([]);
+  };
+
+  const [isOpen, setOpen] = useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setSelectedDates([]);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = async () => {
+    // Use YYYY-MM-DD format
+    const dateRange = selectedDates.map((date) =>
+      date.toISOString().substring(0, 10),
```

Comment:
One thing to call out, `toISOString()` converts the date to UTC (in our case, 5 hours ahead). Just want to ensure this isn't going to cause any weird date issues (only reason I know this is because I spent like a month at my past co-op trying to debug something like this lol)

Reply:
I can make sure the date is always in the same time zone as Toronto by doing this instead:
```
date.toLocaleString("fr-CA", {timeZone: "America/Toronto"}).substring(0, 10)
```

Would this fix the issue you are mentioning?

**Note:** "fr-CA" sets the format to yyyy-MM-dd

Reply:
Should be good! This is only being used in Waterloo so it's safe to do that